### PR TITLE
Add options to disable

### LINKED ISSUE #5: Add option to disable line numbers

### PR DESCRIPTION
**Pull request by nakulcg**
* line number
* git modification marker
* file name

fix #5

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -350,6 +363,24 @@ fn run() -> Result<()> {
...
+        .arg(
+            Arg::with_name("disable-line-number")
+                .short("n")
+                .long("disable-line-number")
+                .help("Disable line number"),
+        )
+        .arg(
+            Arg::with_name("disable-git-modfication-marker")
+                .short("g")
+                .long("disable-git-modfication-marker")
```
Comment (sharkdp) [inline src/main.rs]: typo: modfication => modification. Maybe just `disable-git-markers`?

File: src/main.rs [hunk 2]
```diff
@@ -350,6 +363,24 @@ fn run() -> Result<()> {
...
+        .arg(
+            Arg::with_name("disable-file-name")
+                .short("f")
+                .long("disable-file-name")
+                .help("Disable file name"),
+        )
+        .arg(
+            Arg::with_name("disable-line-number")
+                .short("n")
+                .long("disable-line-number")
```
Comment (sharkdp) [inline src/main.rs]: maybe `disable-line-numbers`?

File: src/main.rs [hunk 3]
```diff
@@ -350,6 +363,24 @@ fn run() -> Result<()> {
                 .multiple(true)
                 .empty_values(false),
         )
+        .arg(
+            Arg::with_name("disable-file-name")
+                .short("f")
+                .long("disable-file-name")
```
Comment (sharkdp) [inline src/main.rs]: maybe `disable-file-names`?

**Comment by nakulcg** [comment]
Thanks for such a quick feedback. Sure I would say 
* We can either retain my additions with the small changes you have suggested
* We can have a single clap arg like as you suggested with below possible values
** plain/minimal - only content, no  horizontal separators above and below, no left sided vertical lines
** numbers - only numbers
** full - numbers, and markers

In addition we can change filename to only show if multiple files

What do you think? I am leaning for the 2nd one

**Comment by sharkdp** [comment]
> We can have a single clap arg like as you suggested with below possible values
> plain/minimal - only content
> numbers - only numbers
> full - numbers, and markers

Sounds great.

> We can change filename to only show if multiple files

I thouhgt about this again.. the header bar with the file name is not super-helpful for a single file right now, but I was planning to add some more information to that line (git status, number of added/deleted/modified lines, last modified by, etc.). We should definitely disable the header for `--style=plain`, but I think I'd like to keep it enabled by default, for now.
 
> In addition we can show the horizontal lines only in case of multiple files

If we keep the header, I'd also like to keep the horizontal lines for now. Again, they should definitely be removed for `--style=plain`. Maybe we could later add some `--grid` option that would allow us to set the style for all grid lines (off, only vertical lines, plain ASCII characters instead of Unicode block-drawing symbols, ...).

Thoughts?

**Comment by nakulcg** [comment]
Sure. Then lets do the style option with
* plain - no line numbers and no modification markers
* line-numbers - only line numbers
* full - line numbers and markers for now, and in future other things

**Comment by sharkdp** [comment]
Thank you!

**Comment by sharkdp** [comment]
Oh, for some reason I thought it would completely disable the horizontal/vertical bars for `--style=plain`.

**Comment by sharkdp** [comment]
> and retained it as we wanted to retain the name anyway

I think "plain" should really just be the file contents, nothing else. Sorry if I confused you with the file-name/header thing.

> and with multiple files this would anyway be an issue.

Why? It would be the same as the normal `cat` behavior. The content of the different files would be concatenated.


Would you agree with these statements? If yes, would you like to address this in another PR? Otherwise, I can also implement this myself.

In any case, thanks!

**Pull request merged**